<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="utf-8"/>
	<title>bedford lab / blog</title>
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	
	<!-- RSS feed -->
	<link rel="alternate" type="application/rss+xml" title="Bedford Lab Research" href="http://laduplessis.github.io//feed.xml">
	
    <!-- Customized Bootstrap + Font Awesome + Solarized -->
    <link href="/blotter/css/style.css" rel="stylesheet" media="screen">

	<!-- Favicon -->
	<link rel="shortcut icon" href="/blotter/images/favicon.png"/>	

	<!-- Typekit -->
	<script>
	  (function(d) {
		var config = {
		  kitId: 'xeu8jut',
		  scriptTimeout: 3000
		},
		h=d.documentElement,t=setTimeout(function(){h.className=h.className.replace(/\bwf-loading\b/g,"")+" wf-inactive";},config.scriptTimeout),tk=d.createElement("script"),f=false,s=d.getElementsByTagName("script")[0],a;h.className+=" wf-loading";tk.src='//use.typekit.net/'+config.kitId+'.js';tk.async=true;tk.onload=tk.onreadystatechange=function(){a=this.readyState;if(f||a&&a!="complete"&&a!="loaded")return;f=true;clearTimeout(t);try{Typekit.load(config)}catch(e){}};s.parentNode.insertBefore(tk,s)
	  })(document);
	</script>
		
	<!-- Google Analytics -->	
	<script>
		(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
		(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
		m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
		})(window,document,'script','//www.google-analytics.com/analytics.js','ga');

		ga('create', 'UA-26244371-2', 'laduplessis.github.io/Taming_the_BEAST_blotter');
		ga('send', 'pageview');

	</script>
	
	<!-- jQuery -->
	<script src="/blotter/js/jquery.min.js"></script>
	
	<!-- Bootstrap -->
	<script src="/blotter/js/transition.js"></script>
	<script src="/blotter/js/collapse.js"></script>

</head>

<body>
	
	<div id="header">
	<nav class="navbar navbar-default navbar-static-top" role="navigation">	 
		<div class="container">	

			<div class="navbar-header">
				<!--button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar-collapse-element" aria-expanded="false">
					<span class="sr-only">Toggle navigation</span>
					<span class="icon-bar"></span>
					<span class="icon-bar"></span>
					<span class="icon-bar"></span>
				</button-->
									
				<a class="navbar-brand" href="/blotter/">Taming the BEAST</a>
				<object class="logo navbar-brand" data="/blotter/images/logo_bw.svg" type="image/svg+xml"></object>
				
			</div>

			<div class="collapse navbar-collapse" id="navbar-collapse-element">
				<ul class="nav navbar-nav navbar-right">
					
					<li class="active">
					
					<a href="/blotter/blog/">blog</a>
					</li>
					
					<li>
					
					<a href="/blotter/projects/">projects</a></li>	  
					
					<li>
					
					<a href="/blotter/team/">team</a></li>	  	  
				</ul>
			</div>
			
		</div>
	</nav>	
</div>
	
	<div class="container">	
		
	<div class="row">
	<div class="col-md-12">
		<div class="centered-pills">
			<ul class="nav nav-pills note">
				<li role="presentation" class="active">
					<a class="off" href="/blotter/blog/">
						<i class="fa fa-refresh fa-fw"></i> Timeline
					</a>
				</li>
				<li role="presentation">
					<a class="off" href="/blotter/blog/archive/">
						<i class="fa fa-list-ul fa-fw"></i> Index
					</a>
				</li>
			</ul>
		</div>
		<div class="bigspacer"></div>
	</div>
</div>

<div class="post-list">



	

	<div class="blog-post">
		<div class="row">
			<div class="col-md-1"></div>	
			<div class="col-md-10">
				<div class="media">
					<div class="media-body">
						<div class="title media-heading">
							<a href="/blotter/blog/zibra-project/" class="off">Zika sequencing in Brazil (or Trevor does fieldwork)</a>
						</div>
						<p>
						<div class="smallhead">
							10 Jun 2016
							
							by <a class="off" href="/blotter/team/trevor-bedford/">trvrb</a>
									
						</div>
					</div>
				</div>
			</div>
			<div class="col-md-1"></div>					
		</div>

		<div class="bigspacer"></div>

		<div class="row">
			<div class="col-md-1"></div>
			<div class="col-md-10">	
				
				
				<a href="/blotterhttp://zibraproject.github.io">
					<img class="offset" src="/blotter/images/blog/zibra_logo.png">
				</a>			
								
				
				<div class="post">
					<p>I&#39;m at the Rio airport now, heading home after 9 days in Brazil as part of the ground team of the <a href="http://zibraproject.github.io/about/">ZiBRA project</a>. As part of the team, I traveled from Natal to Recife along the northeastern coast collecting clinical samples for mobile Zika genome sequencing and analysis. This has been an illuminating experience and I&#39;m grateful to <a href="http://lab.loman.net/">Nick</a>, <a href="http://evolve.zoo.ox.ac.uk/Evolve/Nuno_Faria.html">Nuno</a>, <a href="http://www.pgpat.ufba.br/docenteDetalhes.asp?id=FS0K6QkMSK">Luiz</a> and the <a href="http://zibraproject.github.io/people/">rest of the team</a> for inviting me to be part of this.</p>

<p>I truly believe that pathogen genome analysis can contribute significantly to epidemiological understanding and outbreak response. However, for this to work, genomes need to be produced and shared quickly enough so that epidemiological insights are <em>actionable</em>. <a href="/blog/scientific-publishing-practices/">This was a major issue for much of the West African Ebola outbreak</a>, limiting the utility of genomic approaches. The situation is somewhat better for the ongoing Zika epidemic in the Americas in that multiple groups are releasing a genome here and a genome there, but overall depth is still lacking with just 64 outbreak genomes available at this time. The ZiBRA project is an attempt to do real-time genomic surveillance of Zika in Brazil. If all goes according to plan, this project will rapidly provide a dataset for downstream analysis of Zika evolution and epidemiology, aiding understanding of virus spread and epidemic dynamics.</p>

<p>The trip was incredibly eye-opening for me in terms of the messy reality of viral surveillance and the even-more-messy details of Zika surveillance in Brazil. The basic pipeline for Zika surveillance in Brazil by the Ministry of Health (much like other viral surveillance systems) goes something like:</p>

<ol>
<li>Patient presents at a clinic with symptoms consistent with infection (fever, rash, etc...).</li>
<li>The clinician sends a blood sample to the regional diagnostic laboratory (these are referred to as LACENs).</li>
<li>The LACEN extracts viral RNA and runs <a href="https://en.wikipedia.org/wiki/Real-time_polymerase_chain_reaction">RT-PCR</a> to confirm viral presence in the sample.</li>
</ol>

<p>The RT diagnostic is particularly important as clinical symptoms are difficult to distinguish between Zika, dengue and Chikungunya.  With the <a href="http://zibraproject.github.io/roadtrip/">road trip</a>, we were able to bring in reagents and expertise lacked by the LACENs and burn through a large number of banked clinical specimens to search for additional RT-positives. In some cases, we were able to confirm Zika diagnoses of pregnant women who presented the week before. We reported postive and negative RT diagnostics back to the LACENs. RT-positive samples were then brought forward for <a href="http://zibraproject.github.io/blog/multiplex-pcr-protocol/">PCR amplification</a> and <a href="http://zibraproject.github.io/blog/protocol-low-input-native-barcoding-protocol/">MinION sequencing</a>.</p>

<p>I did help a bit with the lab-work, but I ended up mostly running point on metadata. As might be expected given the circumstances, the lab work was incredibly chaotic and I spent most of my time trying to keep sample data from <em>unraveling</em>. To keep epi metadata attached to a sample required maintaining a linkage between the numbers written from tube-to-tube-to-tube and the original LACEN ID. It also required digging through the LACEN diagnostic reports to pull in important epi metadata like date of collection and municipality of residence. I&#39;ve never quite appreciated before the degree to which data wants to come apart if continual attention is not paid (proper data is an ordered state that is constantly under attack by entropic forces). I hope I&#39;ve left the team with <a href="http://zibraproject.github.io/blog/metadata-wrangling/">systems in place to promote further metadata collection</a>.</p>

<p>We finished base calling and assembly on the first MinION runs on June 8, but realized they need resequencing to have good coverage. That said, we should be releasing genomes soon and hope to keep a flow of genomes going through the next few months. I&#39;m super excited to be able to rapidly incorporate these genomes into <a href="http://nextstrain.org/zika/">nextstrain.org</a> and help with tracking Zika evolution and epidemic spread.</p>

<div class="row">
    <div class="col-sm-2">
        <a href="/images/blog/zibra_photos/img_1183.jpg">
            <img src="/images/blog/zibra_photos_thumbnails/img_1183.jpg">
        </a>
        <div class="bigspacer"></div>
    </div>
    <div class="col-sm-2">
        <a href="/images/blog/zibra_photos/img_1190.jpg">
            <img src="/images/blog/zibra_photos_thumbnails/img_1190.jpg">
        </a>
        <div class="bigspacer"></div>
    </div>
    <div class="col-sm-2">
        <a href="/images/blog/zibra_photos/img_1210.jpg">
            <img src="/images/blog/zibra_photos_thumbnails/img_1210.jpg">
        </a>
        <div class="bigspacer"></div>
    </div>
    <div class="col-sm-2">
        <a href="/images/blog/zibra_photos/img_1266.jpg">
            <img src="/images/blog/zibra_photos_thumbnails/img_1266.jpg">
        </a>
        <div class="bigspacer"></div>
    </div>
    <div class="col-sm-2">
        <a href="/images/blog/zibra_photos/img_1434.jpg">
            <img src="/images/blog/zibra_photos_thumbnails/img_1434.jpg">
        </a>
        <div class="bigspacer"></div>
    </div>
    <div class="col-sm-2">
        <a href="/images/blog/zibra_photos/img_1481.jpg">
            <img src="/images/blog/zibra_photos_thumbnails/img_1481.jpg">
        </a>
        <div class="bigspacer"></div>
    </div>  
</div>

<div class="row">
    <div class="col-sm-2">
        <a href="/images/blog/zibra_photos/img_1541.jpg">
            <img src="/images/blog/zibra_photos_thumbnails/img_1541.jpg">
        </a>
        <div class="bigspacer"></div>
    </div>
    <div class="col-sm-2">
        <a href="/images/blog/zibra_photos/img_1577.jpg">
            <img src="/images/blog/zibra_photos_thumbnails/img_1577.jpg">
        </a>
        <div class="bigspacer"></div>
    </div>
    <div class="col-sm-2">
        <a href="/images/blog/zibra_photos/img_1582.jpg">
            <img src="/images/blog/zibra_photos_thumbnails/img_1582.jpg">
        </a>
        <div class="bigspacer"></div>
    </div>
    <div class="col-sm-2">
        <a href="/images/blog/zibra_photos/img_1606.jpg">
            <img src="/images/blog/zibra_photos_thumbnails/img_1606.jpg">
        </a>
        <div class="bigspacer"></div>
    </div>
    <div class="col-sm-2">
        <a href="/images/blog/zibra_photos/img_1629.jpg">
            <img src="/images/blog/zibra_photos_thumbnails/img_1629.jpg">
        </a>
        <div class="bigspacer"></div>
    </div>
    <div class="col-sm-2">
        <a href="/images/blog/zibra_photos/img_1632.jpg">
            <img src="/images/blog/zibra_photos_thumbnails/img_1632.jpg">
        </a>
        <div class="bigspacer"></div>
    </div>
</div>

<div class="spacer"></div>

				</div>	
			</div>			
			<div class="col-md-1"></div>
		</div>
	
		<div class="bigspacer"></div>
		<div class="spacer"></div>	

	</div>



	

	<div class="blog-post">
		<div class="row">
			<div class="col-md-1"></div>	
			<div class="col-md-10">
				<div class="media">
					<div class="media-body">
						<div class="title media-heading">
							<a href="/blotter/blog/programmer-real-time-analysis/" class="off">Programmer for real-time phylodynamic analysis</a>
						</div>
						<p>
						<div class="smallhead">
							4 Apr 2016
							
							by <a class="off" href="/blotter/team/trevor-bedford/">trvrb</a>
									
						</div>
					</div>
				</div>
			</div>
			<div class="col-md-1"></div>					
		</div>

		<div class="bigspacer"></div>

		<div class="row">
			<div class="col-md-1"></div>
			<div class="col-md-10">	
				
				<div class="post">
					<p>We&#39;re looking for a programmer to help with the current push towards real-time analysis of virus evolution. The advertisement follows:</p>

<blockquote>
<p>A programmer position is available immediately in the Bedford lab at the Fred Hutch to develop inference algorithms and interactive visualizations of viral outbreaks using DNA/RNA sequence data.</p>

<p>There has been remarkable progress in <a href="https://en.wikipedia.org/wiki/Viral_phylodynamics">phylodynamic</a> methods which use viral genetic sequence data to infer patterns of epidemic growth and geographic spread as well as patterns of adaptive evolution and strain turnover. However, until recently, these methods were solely applied in retrospective analyses to understand past events. With increasing availability and timeliness of sequence data, it is now becoming possible to perform and share phylodynamic analyses in near real-time. </p>

<p>Our group is leading this transition to real-time analysis and prediction. Current efforts by the lab include analyses of influenza (<a href="http://nextflu.org">nextflu.org</a>), Ebola (<a href="http://ebola.nextstrain.org">ebola.nextstrain.org</a>) and Zika (<a href="http://nextstrain.org/zika">nextstrain.org/zika/</a>) viruses. These sites are already being used by the Centers for Disease Control and the World Health Organization, particularly for influenza vaccine strain selection, and we believe further development will lead to substantial public health benefit. This project has grown to the point where we need a full-time programmer to generalize, refactor, and maintain our platform for data ingestion, processing and visualization.</p>

<p>The ideal candidate would have experience in Python (informatic processing pipeline <a href="https://github.com/blab/nextflu/tree/master/augur">augur</a> is written in Python) and Javascript (browser-based visualization <a href="https://github.com/blab/nextflu/tree/master/auspice">auspice</a> is written in Javascript). Experience with databases (we are using <a href="http://www.rethinkdb.com/">rethinkdb</a> for sequence organization) or web application programming would be a plus. The new team member would contribute to refactor the current codebase to make a more general and extensible platform for informatic processing and also to speed up the auspice visualization by reactive JSON pull-downs. Additional development would focus on extending pipelines to other viruses / pathogens and building new features for current viruses, as well as, implementing a robust <a href="https://www.docker.com/">Docker</a>-based deployment pipeline.</p>

<p>The Fred Hutch is located in South Lake Union in Seattle, WA and offers a dynamic work environment with cutting-edge science and computational resources. The position is available immediately with flexible starting dates. Informal inquires are welcome. Applications will be accepted until the position is filled.  We offer a competitive salary commensurate with skills and experience, along with benefits. The Fred Hutch and the Bedford lab are committed to improving diversity in the computational sciences. Applicants of diverse backgrounds are particularly encouraged to apply.</p>

<p>For more information about the lab, please see the our website at <a href="http://bedford.io">bedford.io</a>. To apply for the position please send (1) current resume, (2) code samples or links to published/distributed code and (3) contact information for two references to trevor<span style="display:none">obfuscate</span>@bedford.io.</p>
</blockquote>

				</div>	
			</div>			
			<div class="col-md-1"></div>
		</div>
	
		<div class="bigspacer"></div>
		<div class="spacer"></div>	

	</div>



	

	<div class="blog-post">
		<div class="row">
			<div class="col-md-1"></div>	
			<div class="col-md-10">
				<div class="media">
					<div class="media-body">
						<div class="title media-heading">
							<a href="/blotter/blog/feb-2016-nextflu-report/" class="off">Seasonal influenza circulation patterns and future projections</a>
						</div>
						<p>
						<div class="smallhead">
							15 Feb 2016
							
							by <a class="off" href="/blotter/team/trevor-bedford/">trvrb</a>
									
						</div>
					</div>
				</div>
			</div>
			<div class="col-md-1"></div>					
		</div>

		<div class="bigspacer"></div>

		<div class="row">
			<div class="col-md-1"></div>
			<div class="col-md-10">	
				
				
				<a href="/blotterhttp://nextflu.org/reports/feb-2016/">
					<img class="offset" src="/blotter/images/blog/h3n2_clade_trajectory.png">
				</a>			
								
				
				<div class="post">
					<p><a href="https://neherlab.wordpress.com/">Richard Neher</a> and I have compiled a <a href="http://nextflu.org/reports/feb-2016/">report on recent patterns of seasonal influenza virus evolution</a> with projections for future clade behavior based on the <a href="http://nextflu.org">nextflu</a> platform. This is designed to be looking forward to the 2016-2017 flu season. Lots of interesting behavior in H3N2 and H1N1pdm viruses. Within H3N2, the clade 3c2.a has continued to predominate throughout the last year and now genetic diversity is accumulating within 3c2.a. The issue is now watching for new variants within 3c2.a that may be antigenically distinct and thus spread through the virus population. H1N1pdm viruses have seen the rapid rise a novel clade denoted 6b.2 during 2015. This clade is demarcated by the substitutions 84N/162N/216T. The joint substitutions 162N/216T have risen especially rapidly with this season&#39;s Northern Hemisphere epidemic comprised of almost entirely 6b.2 viruses. However, detailed antigenic characterization is necessary to determine whether the seasonal influenza vaccine warrants updating.</p>

				</div>	
			</div>			
			<div class="col-md-1"></div>
		</div>
	
		<div class="bigspacer"></div>
		<div class="spacer"></div>	

	</div>



	

	<div class="blog-post">
		<div class="row">
			<div class="col-md-1"></div>	
			<div class="col-md-10">
				<div class="media">
					<div class="media-body">
						<div class="title media-heading">
							<a href="/blotter/blog/rtss-sieve/" class="off">Strain-specific vaccine efficacy in the RTS,S malaria vaccine</a>
						</div>
						<p>
						<div class="smallhead">
							26 Oct 2015
							
							by <a class="off" href="/blotter/team/trevor-bedford/">trvrb</a>
									
						</div>
					</div>
				</div>
			</div>
			<div class="col-md-1"></div>					
		</div>

		<div class="bigspacer"></div>

		<div class="row">
			<div class="col-md-1"></div>
			<div class="col-md-10">	
				
				
				<a href="/blotter/papers/neafsey-rtss-sieve/">
					<img class="offset" src="/blotter/images/blog/rtss_sieve.png">
				</a>			
								
				
				<div class="post">
					<p>In collaboration with <a href="https://www.broadinstitute.org/bios/daniel-neafsey">Dan Neafsey</a> at the Broad Institute, <a href="http://www.hsph.harvard.edu/dyann-wirth/">Dyann Wirth</a> at Harvard, <a href="https://www.fredhutch.org/en/labs/profiles/gilbert-peter.html">Peter Gilbert</a> and <a href="https://www.fredhutch.org/en/labs/profiles/juraska-michal.html">Michal Juraska</a> at the Fred Hutch and a large team of researchers, we&#39;ve just published <a href="/papers/neafsey-rtss-sieve/">a paper</a> in the <a href="http://dx.doi.org/10.1056/NEJMoa1505819">New England Journal of Medicine</a> showing strain-specific vaccine efficacy in a recent malaria vaccine trial. We found that the <a href="https://en.wikipedia.org/wiki/RTS,S">RTS,S malaria vaccine</a> worked better against strains that were genetically matched to the vaccine antigen compared to unmatched strains. The right hand figure shows that across study sites, parasites matched to the vaccine strain (3D7) are overrepresented among the control group, indicating that the vaccine better protected against infection by 3D7 than against other strains. Proper statistical analysis shows that there was a 1-year vaccine efficacy of 50% against strains perfectly matched to the vaccine antigen and a 1-year vaccine efficacy of 33% against unmatched strains. The vaccine strain 3D7 is at low (~10%) frequency in Sub-Saharan Africa. This suggests that vaccine efficacy could be straight-forwardly improved by just swapping 3D7 for a more common haplotype.</p>

<p>It&#39;s interesting to see strain-specific vaccine for malaria, suggesting that like flu, a good vaccine requires matching to the circulating pathogen strains. This was a fun study to participate in. Hopefully, we could see further improvements to the RTS,S vaccine. Even at current efficacy levels, RTS,S is estimated to have a cost effectiveness of <a href="http://www.malariajournal.com/content/13/1/66">~$150 per disability-adjusted life year</a> saved.</p>

				</div>	
			</div>			
			<div class="col-md-1"></div>
		</div>
	
		<div class="bigspacer"></div>
		<div class="spacer"></div>	

	</div>



	

	<div class="blog-post">
		<div class="row">
			<div class="col-md-1"></div>	
			<div class="col-md-10">
				<div class="media">
					<div class="media-body">
						<div class="title media-heading">
							<a href="/blotter/blog/visualizing-antigenic-evolution/" class="off">Analysis and visualization of influenza antigenic evolution</a>
						</div>
						<p>
						<div class="smallhead">
							6 Oct 2015
							
							by <a class="off" href="/blotter/team/trevor-bedford/">trvrb</a>
									
						</div>
					</div>
				</div>
			</div>
			<div class="col-md-1"></div>					
		</div>

		<div class="bigspacer"></div>

		<div class="row">
			<div class="col-md-1"></div>
			<div class="col-md-10">	
				
				
				<a href="/blotterhttp://HI.nextflu.org">
					<img class="offset" src="/blotter/images/blog/titer_tree.png">
				</a>			
								
				
				<div class="post">
					<p><a href="https://neherlab.wordpress.com/">Richard Neher</a> and <a href="http://www.kitp.ucsb.edu/shraiman/">Boris Shraiman</a>, along with myself, <a href="http://www.vet.cam.ac.uk/directory/russell">Colin Russell</a> and <a href="http://www.crick.ac.uk/research/worldwide-influenza-centre/">Rod Daniels</a> have just completed a new analysis of antigenic drift in the seasonal influenza virus. We&#39;ve put a manuscript on <a href="/papers/neher-titers/">&quot;Prediction, dynamics, and visualization of antigenic phenotypes of seasonal influenza viruses&quot;</a> up on the <a href="http://arxiv.org/abs/1510.01195">arXiv</a> and have created an interactive visualization of our model at <a href="http://HI.nextflu.org">HI.nextflu.org</a>.</p>

<p>Much previous work on this topic, <a href="/papers/bedford-flux/">including my own</a>, has focused on &quot;antigenic cartography&quot; in which antigenic phenotype is embedded into a 2D antigenic map where distances between viruses are proportional to drop in titer in the hemagglutination inhibition (HI) assay. Here, we take a complementary approach where drop in titer is directly mapped to the influenza phylogeny; specific branches are assigned drops in titer. Thus, rather than embedding into 2D Euclidean space, this model assumes embedding into tree space. As a model, it works quite well, allowing prediction of unmeasured titers at high accuracy. In the manuscript, we investigate the correlation between antigenic advancement and clade success, finding that more advanced clades tend to win in the global competition among viruses. However, we observe significant noise when the new clade is at very low frequency.</p>

<p>A major feature of the cartography-based approach is providing a single pictorial view of the antigenic relationships among viruses. This is enabled by the 2D basis for cartography. The tree-based model that we present here does not permit a single viewpoint. Instead, we provide an <a href="http://HI.nextflu.org">interactive visualization</a> of the model, in which users can click on a particular reference virus in the phylogeny and see how other viruses relate to this focal virus. We allow coloring of the phylogeny by raw HI titer, as well as by model expectations, allowing exploration of <em>why</em> the model behaves as it does. We hope this approach will be useful in investigation of antigenic relationships among circulating influenza viruses.</p>

				</div>	
			</div>			
			<div class="col-md-1"></div>
		</div>
	
		<div class="bigspacer"></div>
		<div class="spacer"></div>	

	</div>



	

	<div class="blog-post">
		<div class="row">
			<div class="col-md-1"></div>	
			<div class="col-md-10">
				<div class="media">
					<div class="media-body">
						<div class="title media-heading">
							<a href="/blotter/blog/sep-2015-nextflu-report/" class="off">Seasonal influenza circulation patterns and future projections</a>
						</div>
						<p>
						<div class="smallhead">
							24 Sep 2015
							
							by <a class="off" href="/blotter/team/trevor-bedford/">trvrb</a>
									
						</div>
					</div>
				</div>
			</div>
			<div class="col-md-1"></div>					
		</div>

		<div class="bigspacer"></div>

		<div class="row">
			<div class="col-md-1"></div>
			<div class="col-md-10">	
				
				<div class="post">
					<p><a href="https://neherlab.wordpress.com/">Richard Neher</a> and I have compiled a <a href="http://nextflu.org/reports/sep-2015/">report on recent patterns of seasonal influenza virus evolution</a> with projections for future clade behavior. The <a href="http://nextflu.org">nextflu</a> site made putting this together really straightforward. Lots of interesting stuff going on. The 3c2.a clade is taking over H3N2 viruses, there&#39;s a new clade of H1N1pdm viruses localized in South Asia and B viruses continue to oscillate between competing clades without permanent winners. </p>

				</div>	
			</div>			
			<div class="col-md-1"></div>
		</div>
	
		<div class="bigspacer"></div>
		<div class="spacer"></div>	

	</div>



	

	<div class="blog-post">
		<div class="row">
			<div class="col-md-1"></div>	
			<div class="col-md-10">
				<div class="media">
					<div class="media-body">
						<div class="title media-heading">
							<a href="/blotter/blog/sismid-course/" class="off">Pathogen evolution, selection and immunity</a>
						</div>
						<p>
						<div class="smallhead">
							20 Jul 2015
							
							by <a class="off" href="/blotter/team/trevor-bedford/">trvrb</a>
									
						</div>
					</div>
				</div>
			</div>
			<div class="col-md-1"></div>					
		</div>

		<div class="bigspacer"></div>

		<div class="row">
			<div class="col-md-1"></div>
			<div class="col-md-10">	
				
				
				<a href="/blotter/projects/sismid/">
					<img class="offset" src="/blotter/images/blog/wright_fisher.png">
				</a>			
								
				
				<div class="post">
					<p>This week, <a href="http://cobeylab.uchicago.edu/">Sarah Cobey</a> and I are co-teaching a short course on &quot;Pathogen evolution, selection and immunity&quot; for the 2015 <a href="http://sismid.uw.edu">Summer Institute in Statistics and Modeling in Infectious Diseases (SISMID)</a>. This course is designed as an introduction to epidemiological modeling and evolutionary analysis for antigenically variable pathogens. Here, on the modeling side, we cover immune-mediated competition, multistrain mechanistic models and state-space reconstruction, while on the evolution side we cover the coalescent, phylogenetics and natural selection. We also discuss the immune response in terms of serology and B cell dynamics. I&#39;m very excited about co-teaching this course.</p>

<p>Sarah and I have worked together to put all the course materials online. The course website is <a href="/projects/sismid/">here</a>. We include slide decks using <a href="https://github.com/hakimel/reveal.js/">Reveal.js</a> for individual topics like <a href="/projects/sismid/competition/slides.html">competition</a> and the <a href="/projects/sismid/coalescent/slides.html">coalescent</a>. We also include introductory exercises for topics like <a href="/projects/sismid/phylogenetics/practical/">phylogenetics using BEAST</a> and <a href="/projects/sismid/ssr/exercise/">state-space reconstruction</a>. The goal was to have something students could view alongside lectures, but also have something to refer to after the course was finished and to link out to follow-up material. Overall, I&#39;m really quite pleased with this set up.</p>

				</div>	
			</div>			
			<div class="col-md-1"></div>
		</div>
	
		<div class="bigspacer"></div>
		<div class="spacer"></div>	

	</div>



	

	<div class="blog-post">
		<div class="row">
			<div class="col-md-1"></div>	
			<div class="col-md-10">
				<div class="media">
					<div class="media-body">
						<div class="title media-heading">
							<a href="/blotter/blog/new-recruits/" class="off">New recruits</a>
						</div>
						<p>
						<div class="smallhead">
							30 Jun 2015
							
							by <a class="off" href="/blotter/team/trevor-bedford/">trvrb</a>
									
						</div>
					</div>
				</div>
			</div>
			<div class="col-md-1"></div>					
		</div>

		<div class="bigspacer"></div>

		<div class="row">
			<div class="col-md-1"></div>
			<div class="col-md-10">	
				
				<div class="post">
					<p>The lab is suddenly growing. We&#39;ve had 4 new people join in the last ~2 weeks. <a href="/team/allison-black/">Allison Black</a> is starting a UW Epidemiology PhD in the fall, <a href="/team/gabe-erion/">Gabe Erion</a> is doing a summer rotation with the group as part of the UW MD/PhD program, <a href="/team/charlton-callender/">Charlton Callender</a> is an undergraduate at the UW doing a summer research internship as part of the Fred Hutch SURP program and <a href="/team/sidney-bell/">Sidney Bell</a> is starting a 1 year UW Epidemiology Master&#39;s adjunct to the MCB PhD program. This should be a fun summer with lots of projects to hack on together. I&#39;m very much looking forward to it.</p>

				</div>	
			</div>			
			<div class="col-md-1"></div>
		</div>
	
		<div class="bigspacer"></div>
		<div class="spacer"></div>	

	</div>



</div>

<div class="infinite-spinner"></div>

<div class="row">
	<div class="col-md-12">
		<div id="scroll" class="centered-pills">
			<ul class="nav nav-pills note">
				<li role="presentation">
					<a class="off" href="/blotter/blog/">
						<i class="fa fa-chevron-up fa-fw"></i> Back to top
					</a>
				</li>				
			</ul>
		</div>
	</div>
</div>

<!-- Infinite scroll -->	
<script async src="/blotter/js/infinite-scroll.js"></script>
	
	</div>
	
	<div id="footer"><span style="display:none">foo</span></div>
		
</body>
</html>

